backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v6.ts

```diff
@@ -0,0 +1,922 @@
+    }
+
+    const upsertResult = await this.experimentUserService.upsertOnChange(
+      experimentUserDoc,
+      experimentUser,
+      request.logger
+    );
+
+    if (!upsertResult) {
+      request.logger.error({
+        details: 'user upsert failed',
+      });
+      throw new InternalServerError('user init failed');
+    }
+
+    return { id, group, workingGroup };
+  }
+
+  /**
+   * @swagger
+   * /v6/groupmembership:
+   *    patch:
+   *       description: Set group membership for a user
+   *       consumes:
+   *         - application/json
+   *       parameters:
+   *         - in: body
+   *           name: experimentUser
+   *           required: true
+   *           schema:
+   *             type: object
+   *             properties:
+   *               id:
+   *                 type: string
+   *                 example: user1
+   *               group:
+   *                 type: object
+   *                 properties:
+   *                   schoolId:
+   *                      type: array
+   *                      items:
+   *                        type: string
+   *                        example: school1
+   *                   classId:
+   *                      type: array
+   *                      items:
+   *                        type: string
+   *                        example: class1
+   *                   instructorId:
+   *                      type: array
+   *                      items:
+   *                        type: string
+   *                        example: instructor1
+   *           description: ExperimentUser
+   *       tags:
+   *         - Client Side SDK
+   *       produces:
+   *         - application/json
+   *       responses:
+   *          '200':
+   *            description: Set Group Membership
+   *            schema:
+   *              $ref: '#/definitions/v6initResponse'
+   *          '400':
+   *            description: BadRequestError - InvalidParameterValue
+   *          '401':
+   *            description: AuthorizationRequiredError
+   *          '404':
+   *            description: Experiment User not defined
+   *          '500':
+   *            description: Internal Server Error
+   */
+  @Patch('groupmembership')
+  public async setGroupMemberShip(
+    @Req()
+    request: AppRequest,
+    @Body({ validate: true })
+    experimentUser: ExperimentUserValidator
+  ): Promise<IGroupMembership> {
+    request.logger.info({ message: 'Starting the groupmembership call for user' });
+    // getOriginalUserDoc call for alias
+    const experimentUserDoc = await this.experimentUserService.getUserDoc(experimentUser.id, request.logger);
+    if (experimentUserDoc) {
+      // append userDoc in logger
+      request.logger.child({ userDoc: experimentUserDoc });
+      request.logger.info({ message: 'Got the original user doc' });
+    }
+    const { id, group } = await this.experimentUserService.updateGroupMembership(
```

Comment:
Don't fetch the document again. From the middleware attach the document in the req object and use that in the controller.